tests/tizen_nnfw_runtime/unittest_tizen_nnfw_runtime_raw.cpp

```diff
@@ -250,6 +255,163 @@ TEST (nnstreamer_nnfw_runtime_raw_functions, invoke)
+    ASSERT_EQ (ret, 0);
+  }
+
+  const gchar *model_files[] = { model_file, NULL, };
+  GstTensorFilterProperties prop = {
+    .fwname = "nnfw",
+    .fw_opened = 0,
+    .model_files = model_files,
+    .num_models = 1,
+  };
+
+  const GstTensorFilterFramework *sp = nnstreamer_filter_find ("nnfw");
+  EXPECT_NE (sp, (void *) NULL);
+
+  info.num_tensors = 1;
+  info.info[0].type = _NNS_UINT8;
+  info.info[0].dimension[0] = 3;
+  info.info[0].dimension[1] = 224;
+  info.info[0].dimension[2] = 224;
+  info.info[0].dimension[3] = 1;
+
+  ret = sp->open (&prop, &data);
+  EXPECT_EQ (ret, 0);
+
+  /** get input/output dimension successfully */
+  ret = sp->getInputDimension (&prop, &data, &res);
+  EXPECT_EQ (ret, 0);
+
+  EXPECT_EQ (res.num_tensors, info.num_tensors);
+  EXPECT_EQ (res.info[0].type, info.info[0].type);
+  EXPECT_EQ (res.info[0].dimension[0], info.info[0].dimension[0]);
+  EXPECT_EQ (res.info[0].dimension[1], info.info[0].dimension[1]);
+  EXPECT_EQ (res.info[0].dimension[2], info.info[0].dimension[2]);
+  EXPECT_EQ (res.info[0].dimension[3], info.info[0].dimension[3]);
+
+  input.type = res.info[0].type;
+  input.size = gst_tensor_info_get_size (&res.info[0]);
+
+  ret = sp->getOutputDimension (&prop, &data, &res);
+  EXPECT_EQ (ret, 0);
+
+  info.num_tensors = 1;
+  info.info[0].type = _NNS_UINT8;
+  info.info[0].dimension[0] = 1001;
+  info.info[0].dimension[1] = 1;
+  info.info[0].dimension[2] = 1;
+  info.info[0].dimension[3] = 1;
+
+  EXPECT_EQ (res.num_tensors, info.num_tensors);
+  EXPECT_EQ (res.info[0].type, info.info[0].type);
+  EXPECT_EQ (res.info[0].dimension[0], info.info[0].dimension[0]);
+  EXPECT_EQ (res.info[0].dimension[1], info.info[0].dimension[1]);
+  EXPECT_EQ (res.info[0].dimension[2], info.info[0].dimension[2]);
+  EXPECT_EQ (res.info[0].dimension[3], info.info[0].dimension[3]);
+
+  output.type = res.info[0].type;
+  output.size = gst_tensor_info_get_size (&res.info[0]);
+
+  input.data = g_malloc(input.size);
+  output.data = g_malloc(output.size);
+
+  data_file = g_build_filename (root_path, "tests", "test_models", "data",
+      "orange.raw", NULL);
+  fd = open(data_file, O_RDONLY);
+
+  EXPECT_TRUE (fd >= 0);
+  /** Invoke output will not match with fd < 0 - input will be random data */
+  if (fd >= 0) {
+    data_read = read(fd, input.data, input.size);
+    EXPECT_EQ (data_read, input.size);
+    close(fd);
+  }
+
+  ret = sp->invoke_NN (&prop, &data, &input, &output);
+  EXPECT_EQ (ret, 0);
+
+  /** entry 952 (idx 951) is orange as per tests/test_models/labels/labels.txt */
+  max_idx = get_argmax ((guint8 *)output.data, output.size);
+  EXPECT_EQ (max_idx, 951);
+
+  g_free (data_file);
+  g_free (output.data);
+  g_free (input.data);
+  g_free (model_file);
+  sp->close (&prop, &data);
+
+  replace_command = g_strdup_printf("sed -i '/%s/c\\\"models\" : [ \"%s\" ],' %s",
+      new_model, orig_model, manifest_file);
+  ret = system(replace_command);
+  g_free(replace_command);
```

Comment:
It's minor comment: need a space between a function name and its arguments for uniformity.

Reply:
updated.